File: tervezes/uml_gen/uml_generator/MSBear.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace VeryGoodViroGame
{
    public class MSBear : iMoveStrategy
    {
        public void ExecuteMove(Virologist v, Field from, Field to)
        {
            Logger.NewFunctionCall(this, "ExecuteMove");
            Field random = from.GetRandomNeighbour();
            from.RemoveViro(v);
            IList<Virologist> virologists = random.GetVirologists();
            foreach (Virologist viro in virologists)
            {
                viro.ApplyAgent(new Bear(), v);
            }

            random.AcceptViro(v);
            random.Destroy();
            Logger.ReturnFunction();
        }
    }
}

File: VeryGoodViroGame/src/VeryGoodViroGame/MoveStrategy/MSBear.java
package VeryGoodViroGame.MoveStrategy;

import VeryGoodViroGame.Agent.Bear;
import VeryGoodViroGame.EntityManager;
import VeryGoodViroGame.Field.Field;
import VeryGoodViroGame.Logger;
import VeryGoodViroGame.Virologist;

import java.util.List;

public class MSBear implements iMoveStrategy
{
    @Override
    public void ExecuteMove(Virologist v, Field from, Field to)
    {
        Field random = from.GetRandomNeighbour();
        from.RemoveViro(v);
        
        List<Virologist> virologists = random.GetVirologists();
        for(Virologist viro : virologists)
        {
            Bear b = new Bear();
            viro.ApplyAgent(b, v);
            EntityManager.PutCraftedObject("bear", b, "bear_virus");
        }
        
        random.AcceptViro(v);
        
        random.Destroy();
    }
}
